_posts/2023-11-01-#28368.md

```diff
@@ -7,18 +7,79 @@ authors: [ismaelsadeeq]
+
 ## Questions
+
+### Conceptual
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+2. Why is it beneficial to remove `CTxMempool` dependency on `CBlockPolicyEstimator`?
+
+3. Are there any benefits of the `CBlockPolicyEstimator` being a member of `CBlockPolicyEstimator` Are there downsides to removing it?
+
+4. The first attempt [#11775](https://github.com/bitcoin/bitcoin/pull/11775) split the `CValidationInterface` into `CValidationInterface` and `MempoolInterface`. What is the distinction between the two interfaces?
+
+5. Is the split necessary to achieve a background fee estimator? Is it crucial in the long run?
+
+6. Why does this approach omit the split?
+
+7. Can you list the other subscribers to the interface? (Hint: look for classes that inherit from `CValidationInterface`)
+
+8. What does it mean for a class to implement one of the `CValidationInterface` methods? Why is it similar to "subscribing to the event"?
+
+9. What is the difference between implementing a `BlockConnected` and a `NewPoWValidBlock` callback, apart from the fact that they are different events? On which thread is each callback executed?
+
+### Code Review
+
+{:start="10"}
+
+10. In [`4986edb`](https://github.com/bitcoin-core-review-club/bitcoin/commit/4986edb99f8aa73f72e87f3bdc09387c3e516197), Why are we adding a new notification `MempoolTransactionsRemovedForBlock` why can't we use `BlockConnected`?
+
+11. In [`1d116df`](https://github.com/bitcoin-core-review-club/bitcoin/commit/1d116df4c0e021c4c810450e3e5358f34d72940b) a new struct `NewMempoolTransactionInfo` is defined in `kernel/mempool_entry.h` Is that the right place for the struct? What can be a better name for the struct?
+
+12. Are there reasons why the `MempoolTransactionsRemovedForBlock` first parameter is not  `vector<CTxMempoolEntry>`?
+
+13. Are there other alternatives to get the base fee of `CTransactionRef`? Why can't we use it instead of making copies of the base fee in `NewMempoolTransactionInfo`?
+
+
+14. In [`ab4e250`](https://github.com/bitcoin-core-review-club/bitcoin/commit/ab4e250d1d209e0c79dba266461e6b0cfd670452#diff-c065d4cd2398ad0dbcef393c5dfc53f465bf44723348892395fffd2fb3bac522), why is `removeTx` call moved to to reason != `BLOCK` scope? Is it necessary? Is it fixing a bug?
+
+15. Why is the fee estimator not tracking transactions with in-mempool parents?
```

Comment:
Is it not tracking transactions that have in-mempool parents, or is it not tracking those in-mempool parents?

Reply:
Thanks here I used `CTxMempool` variable name for the question, this might be confusing.
updated to 

Why is the fee estimator not tracking transactions with unconfirmed parents?